### DISCUSSION TITLE
Mongodb Event Source?

### DISCUSSION THREAD

**Discussion by justinmchase** [argoproj/argo-events#1197] [Ideas]
Mongodb supports change streams, whenever a record is created/updated/deleted it fires an event through the change stream. It would be super nice if argo could could use mongodb directly as an event source without the need for another system in between the two.

Does anyone know if this is already supported somehow? Or how would I setup a custom event source?

**Comment by whynowy** [comment]
Anybody wants to contribute this feature is welcome!
